File: backend/cinemaservice/internal/domain/schema/auditorium.go
package schema

import (
	"time"

	_ "gorm.io/gorm"
)

type Auditorium struct {
	ID             uint      `gorm:"primaryKey;not null"`
	AuditoriumName string    `gorm:"column:auditorium_name"`
	Capacity       uint      `gorm:"column:capacity"`
	CreatedAt      time.Time `gorm:"column:created_at;autoCreateTime"`
	LastModifiedAt time.Time `gorm:"column:last_modified_at; autoUpdateTime"`
	LastModifiedBy string    `gorm:"column:last_modified_by"`
	CreatedBy      string    `gorm:"column:created_by;type:varchar(50)"`
	IsDeleted      bool      `gorm:"column:is_deleted"`
}

File: frontend/src/modules/admin/auditoriums/models/Auditorium.ts
import { HttpStatusCode } from "axios";

export type Auditorium = {
  auditorium_id: number;
  auditorium_name: string;
  capacity: number;
  is_deleted: boolean;
  created_by: string;
  last_modified_by: string;
  created_at: string;
  last_modified_at: string;
};

export type SearchRequest = {
  page: number;
  pageSize: number;
};

export type Response<T> = {
  status: HttpStatusCode;
  message: string;
  data: T;
};

export type CreateRequest = {
  auditorium_name: string;
  rows: number;
  columns: number;
};
